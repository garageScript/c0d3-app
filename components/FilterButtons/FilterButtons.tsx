import React from 'react'
import { Button } from '../theme/Button'

type FilterProps = {
  options: string[]
  onClick: Function
  currentOption: string
  theme?: 'success' | 'danger' | 'primary' | undefined
}

const FilterButtons: React.FC<FilterProps> = ({
  options,
  onClick,
  currentOption,
  children,
  theme = 'primary'
}) => {
  //incase someone puts in a currentOption value that is not in options array
  if (!options.includes(currentOption)) currentOption = options[0]

  const filters = options.map((option, buttonIndex) => {
    const color = option === currentOption ? 'white' : 'lightgrey'
    const type = option === currentOption ? theme : undefined

    return (
      <div key={buttonIndex} style={{ marginTop: '-1.57px' }}>
        <Button onClick={() => onClick(option)} color={color} btnType={type}>
          {option}
        </Button>
      </div>
    )
  })

  return (
    <div className="d-flex flex-row">
      <h4 className={`text-${theme} fw-bold me-1`}>{children}</h4>
      {filters}
    </div>
  )
}

export default FilterButtons
