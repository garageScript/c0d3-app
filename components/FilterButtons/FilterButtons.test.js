import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import FilterButtons from './FilterButtons'

let testBtnOnClick = ''

const mockOptions = ['Admins', 'Non-Admins', 'None']

describe('FilterButtons Component', () => {
  test('Should call function with updated array as parameter when button is clicked', () => {
    const { container, queryByText } = render(
      <FilterButtons
        currentOption=""
        options={mockOptions}
        onClick={value => (testBtnOnClick = value)}
      >
        Filter By:
      </FilterButtons>
    )
    const button = queryByText('Admins')
    fireEvent.click(button, { button: 1 })
    expect(testBtnOnClick).toEqual('Admins')
    expect(container).toMatchSnapshot()
  })
})
