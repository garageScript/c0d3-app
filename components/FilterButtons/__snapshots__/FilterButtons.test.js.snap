// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButtons Component Should call function with updated array as parameter when button is clicked 1`] = `
<div>
  <div
    class="d-flex flex-row"
  >
    <h4
      class="text-primary fw-bold me-1"
    >
      Filter By:
    </h4>
    <div
      style="margin-top: -1.57px;"
    >
      <button
        class="btn borderless btn-primary"
        style="color: rgb(255, 255, 255);"
      >
        Admins
      </button>
    </div>
    <div
      style="margin-top: -1.57px;"
    >
      <button
        class="btn borderless"
        style="color: rgb(142, 142, 142);"
      >
        Non-Admins
      </button>
    </div>
    <div
      style="margin-top: -1.57px;"
    >
      <button
        class="btn borderless"
        style="color: rgb(142, 142, 142);"
      >
        None
      </button>
    </div>
  </div>
</div>
`;
